File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        longest = 0
        for n in s:
            if n - 1 not in s:
                count = 1
                next_num = n + 1
                while next_num in s:
                    count += 1
                    next_num += 1
                longest = max(longest, count)
        
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Map<Integer, Integer> seqMap = new HashMap<>();
        int maxLen = 0;

        for (int n: nums) {
            if (!seqMap.containsKey(n)) {
                int left = seqMap.getOrDefault(n-1, 0);
                int right = seqMap.getOrDefault(n+1, 0);
                int newLen = left + 1 + right;
                seqMap.put(n, newLen);
                seqMap.computeIfPresent(n-left, (k, len) -> len = newLen);
                seqMap.computeIfPresent(n+right, (k, len) -> len = newLen);

                maxLen = Math.max(maxLen, newLen);
            }
        }
        return maxLen;
    }
}